### ISSUE TITLE
Apollo client support

### ISSUE THREAD

**Issue by ivanornes** [nerdsupremacist/tmdb#4]
Hi, 

I am trying to use the schema.json provided by this API in an iOS Apollo client, but every time I hit run the Apollo script is unable to generate the API.swift file. 

I created and configured an empty Xcode iOS app project with the Apollo client configured and working in this repo https://github.com/ivanornes/Billboard 
The project correctly generated the API.swift file from the [Ray Wenderlich GraphQL and Apollo tutorial](https://www.raywenderlich.com/10886648-graphql-using-the-apollo-framework-getting-started), so I guess that something wrong is with the JSON schema. 

I attach the log that I get in the Xcode console:

```log
[DEBUG - ApolloCodegenLib:CLIDownloader.swift:43] - Zip file with the CLI is already downloaded!
SHASUM of downloaded file: 479e1c12adf631c43c093fdd9dd8e5c8eed33cb15e39de7165ce6f5e7f759904
[DEBUG - ApolloCodegenLib:CLIExtractor.swift:57] - Binary already extracted!
[ERROR - ApolloCodegenLib:Basher.swift:62] -  ›   Warning: apollo update available from 2.31.2 to 2.32.0.
apollo/2.31.2 darwin-x64 node-v10.23.0
+ /Users/ivnrns/xcodeworkspace/ApolloTest/Codegen/ApolloCLI/apollo/bin/run client:download-schema --endpoint=http://127.0.0.1:8080 /Users/ivanornes/xcodeworkspace/ApolloTest/ApolloTest/schema.json
 ›   Warning: apollo update available from 2.31.2 to 2.32.0.
⚠️  It looks like there are 0 files associated with this Apollo Project. This may be because you don't have any files yet, or your includes/excludes fields are configured incorrectly, and Apollo can't find your files. For help configuring Apollo projects, see this guide: https://go.apollo.dev/t/config
Loading Apollo Project [started]
Loading Apollo Project [completed]
Saving schema to /Users/ivanornes/xcodeworkspace/ApolloTest/ApolloTest/schema.json [started]
Saving schema to /Users/ivanornes/xcodeworkspace/ApolloTest/ApolloTest/schema.json [failed]
→ Name "__BaseCreditBasicPerson" must not begin with "__", which is reserved by GraphQL introspection.

Name "__BaseCreditMovie" must not begin with "__", which is reserved by GraphQL introspection.

Name "__BaseCreditMovieOrTV" must not begin with "__", which is reserved by GraphQL introspection.
```

**Comment by nerdsupremacist** [comment]
Ah that rule is new to me. Alright, I'll have a fix soon

**Comment by ivanornes** [comment]
Great! Thanks Mathias! 😊

**Comment by ivanornes** [comment]
Hi Mathias, I have run the tmdb project with the latest changes from the GraphZahl dependency and I see that now the Schema is generating non unique type names. 

```log
[153/153] Linking tmdb
2020-12-25T19:39:30+0100 notice codes.vapor.application : Server starting on http://127.0.0.1:8080
2020-12-25T19:39:40+0100 info codes.vapor.application : request-id=F01CAB07-9D2C-4AB9-A4F1-EA0779B9A1CC GET /
2020-12-25T19:39:42+0100 info codes.vapor.application : request-id=6964531E-4EBF-46CB-8590-FDD8E6F398A8 POST /
2020-12-25T19:39:43+0100 error codes.vapor.application : request-id=6964531E-4EBF-46CB-8590-FDD8E6F398A8 Schema must contain unique named types but contains multiple types named "IBaseCreditBasicPerson".
2020-12-25T19:39:48+0100 info codes.vapor.application : request-id=38F46A18-6D06-4B32-A1F9-C6E646AFDBA6 POST /
2020-12-25T19:39:48+0100 error codes.vapor.application : request-id=38F46A18-6D06-4B32-A1F9-C6E646AFDBA6 Schema must contain unique named types but contains multiple types named "ISeason".
2020-12-25T19:39:49+0100 info codes.vapor.application : request-id=93AAF646-801F-4343-B811-CA5A8EA68912 POST /
2020-12-25T19:39:50+0100 error codes.vapor.application : request-id=93AAF646-801F-4343-B811-CA5A8EA68912 Schema must contain unique named types but contains multiple types named "ITVShow".
2020-12-25T19:39:51+0100 info codes.vapor.application : request-id=8CAFEC01-496B-48D3-9531-9215003139B5 POST /
2020-12-25T19:39:51+0100 error codes.vapor.application : request-id=8CAFEC01-496B-48D3-9531-9215003139B5 Schema must contain unique named types but contains multiple types named "IPerson".
...
```

**Comment by nerdsupremacist** [comment]
Yep those were test commits. It should work now ;)

**Comment by ivanornes** [comment]
Thank you so much for the help Mathias, but still, Apollo doesn't seem to like the final schema. 

I am testing with two projects, one with Apollo's new Codegen project. In the Codegen project I get the following error: 

```log
Loading Apollo Project [started]
Loading Apollo Project [completed]
Generating query files [started]
Generating query files with 'swift' target [title changed]
Generating query files with 'swift' target [failed]
→ Apollo does not support anonymous operations
    GraphQLError: Apollo does not support anonymous operations

Program ended with exit code: 1
```

In the second project I use the old way based on the run-bundled-codegen.sh codegen:generate script. It fails with an error that doesn't tell too much: 

```log
{ Error: Error in "Loading schema for Unnamed Project": TypeError: Cannot read property 'reduce' of undefined
```

**Comment by nerdsupremacist** [comment]
I just looked at the project you told me about and it should work. Do you have any uncommitted changes there?

AFAIK: `Apollo does not support anonymous operations` I think means that your query or mutation doesn't have a name.

Meaning that this cannot be used by Apollo:

```
{
    movie(id: 12) { 
        title
    }
}
```

but instead it should be:

```
query FindingNemoQuery {
    movie(id: 12) { 
        title
    }
}
```

If this isn't the case I can test it tomorrow morning ;)

**Comment by ivanornes** [comment]
I Mathias, I have just updated the schema of the project with the new one. I also have checked the movieList.graphql and it did have the query name.

This is the query: 
```graphql
query MovieList {
  movies {
    popular(first: 10){
      edges {
        node {
          id
          title
          rating
          poster(size:W500)
        }
      }
    }
  }
}
```

I have uploaded the Codegen Project that I am using to [this new repo](https://github.com/ivanornes/GraphQLCodegen).
This repo is the one that generates the "Apollo does not support anonymous operations" logs. 

Thanks Mathias!
